src/atomate2/openmm/utils.py

```diff
@@ -171,3 +172,148 @@ def openff_to_openmm_interchange(
+        """Initialize the reporter.
+
+        Parameters
+        ----------
+        reportInterval : int
+            The interval (in time steps) at which to save frames
+        enforcePeriodicBox : bool | None
+            Whether to wrap coordinates to the periodic box. If None, determined from
+            simulation settings.
+        """
+        self._file = file
+        self._reportInterval = reportInterval
+        self._enforcePeriodicBox = enforcePeriodicBox
+        self._topology = None
+        self._nextModel = 0
+
+        # Storage for trajectory data
+        self._positions: list[np.ndarray] = []
+        self._velocities: list[np.ndarray] = []
+        self._lattices: list[np.ndarray] = []
+        self._frame_properties: list[dict] = []
+        self._species: list[str] | None = None
+        self._time_step: float | None = None
+
+    def describeNextReport(  # noqa: N802
+        self, simulation: Simulation
+    ) -> tuple[int, bool, bool, bool, bool, bool]:
+        """Get information about the next report this object will generate.
+
+        Parameters
+        ----------
+        simulation : Simulation
+            The Simulation to generate a report for
+
+        Returns
+        -------
+        tuple[int, bool, bool, bool, bool, bool]
+            A six element tuple. The first element is the number of steps until the
+            next report. The remaining elements specify whether that report will
+            require positions, velocities, forces, energies, and periodic box info.
+        """
+        steps = self._reportInterval - simulation.currentStep % self._reportInterval
+        return (steps, True, True, False, True, self._enforcePeriodicBox)
+
+    def report(self, simulation: Simulation, state: State) -> None:
+        """Generate a report.
+
+        Parameters
+        ----------
+        simulation : Simulation
+            The Simulation to generate a report for
+        state : State
+            The current state of the simulation
+        """
+        if self._nextModel == 0:
+            self._topology = simulation.topology
+            self._species = [
+                atom.element.symbol for atom in simulation.topology.atoms()
+            ]
+            self._time_step = (
+                simulation.integrator.getStepSize() * self._reportInterval
+            ).value_in_unit(omm_unit.femtoseconds)
+
+        # Get positions and velocities in Angstrom and Angstrom/fs
+        positions = state.getPositions(asNumpy=True).value_in_unit(omm_unit.angstrom)
+        velocities = state.getVelocities(asNumpy=True).value_in_unit(
+            omm_unit.angstrom / omm_unit.femtosecond
+        )
+        box_vectors = state.getPeriodicBoxVectors(asNumpy=True).value_in_unit(
+            omm_unit.angstrom
+        )
+
+        ev_to_kj_per_mol = 96.485
+
+        # Get energies in eV
+        kinetic_energy = (
+            state.getKineticEnergy()._value / ev_to_kj_per_mol  # noqa: SLF001
```

Comment:
is there no public openMM API to get these values? seems fragile to rely on `_value` here which i'm guessing has no backwards compat gurantees?

Reply:
Good catch. I was a bit more clever and fixed it.